scripts/add_directory.js

```diff
@@ -0,0 +1,287 @@
+// set that contains all of the folders currently added as batches
+let batchDirs = new Set();
+// reference to the autocomplete object
+let autocomplete;
+// iterator used to give all batch UI's distinct ID's
+let i = 0;
+// reference to fadeOut timeout for the resultText, used for cancelling the timeout if another result text needs to
+// be shown before the last result text has faded
+let fadeTimeout;
+
+$(function() {    

```

Comment:
Its a jquery thing, it runs the function once the DOM has finished loading. It's similar to ```document.on('load', function() {})``` if you've seen that elsewhere